layer/mounted_layer.go

```diff
@@ -131,8 +131,11 @@ func (rl *referencedRWLayer) Mount(mountLabel string) (string, error) {
 		return "", ErrLayerNotRetained
 	}
 
-	rl.activityCount++
-	return rl.mountedLayer.Mount(mountLabel)
+	m, err := rl.mountedLayer.Mount(mountLabel)
```

Comment:
Not sure I fully understand this logic. Shouldn't we call mount only if `activityCount == 0` and unmount if `activityCount == 1`.